src/coreclr/jit/lsra.h

```diff
@@ -46,7 +47,21 @@ typedef var_types RegisterType;
 template <class T>
 RegisterType regType(T type)
 {
-    return varTypeUsesFloatReg(TypeGet(type)) ? FloatRegisterType : IntRegisterType;
+    if (varTypeUsesIntReg(type))
+    {
+        return IntRegisterType;
+    }
+#if defined(TARGET_XARCH)
+    else if (varTypeUsesMaskReg(type))
+    {
+        return MaskRegisterType;
+    }
+#endif // TARGET_XARCH
+    else
+    {
+        assert(varTypeUsesFloatReg(type));
+        return FloatRegisterType;
+    }
 }
```

Comment:
Checks throughout the JIT have been ordered under the presumption that `varTypeUsesIntReg` is the most likely occurrence. Where mask register handling is needed, `varTypeUsesMaskReg` is the middle case since we only need 1 additional check to cover all 3 cases. This allow the last case to `assert`.

This keeps 1-check for non-xarch, minimizes the total ifdefs when mask handling is needed, and ensures we need no more than 2 checks on xarch with the common case being 1 check. Ideally this helps keep it "pay for play".

Having `varTypeUsesIntReg` allowed a few other checks to become simplified/improved in general.